rust/core-lib/src/movement.rs

```diff
@@ -62,7 +66,7 @@ pub enum Movement {
 /// Note: in non-exceptional cases, this function preserves the `horiz`
 /// field of the selection region.
 fn vertical_motion(r: SelRegion, view: &View, text: &Rope, line_delta: isize,
-    modify: bool) -> (usize, Option<HorizPos>)
+    modify: bool, enforce_horiz_pos: bool) -> (usize, Option<HorizPos>)
```

Comment:
This function is becoming _very_ hard to understand, but I'm not totally sure what the right solution is. I certainly wouldn't want to ever add _another_ bool, after this one, and I'm not sure about this one.

This particular movement feels like it maybe has enough specific logic that it would be worth putting in its own function? I'm not _sure_ this is best, but I'd encourage you to maybe play around with that and see if it simplifies anything?

Reply:
I will play around with it. My problem when looking at it is that so much of what these two methods would have in common is setup, getting positions and lengths and that struck me as odd to put into its own function. Let me play around with it and see what I can do.

Reply:
I've also gone and done and run rustfmt on the whole repo while sleeping on this, so that will be another minor headache. I hope that if you run rustfmt on your branch separately (with the 1.29 compiler, not sure if there are any changes in rustfmt to 1.30) you'll still get a clean merge.

Reply:
I hadn't messed around with rustfmt so it just gave me an excuse to, didn't end up being a problem. Thanks for the heads up.